### ISSUE TITLE
device profile requirements

### ISSUE THREAD

**Issue by george-tsang-san-jose** [CiscoDevNet/yang-explorer#33]
- If netconf IP and other credentials were empty, manage models won't be able to get the device schema since it gets all the access info from the device profile instead of the Build tab. 
- I tried the iosxe platform and it won't even be able to get the device capabilities.

**Comment by pgohite** [comment]
Yes, device profile is required for getting schema from the device. Please use platform 'csr' instead of ios-xe. It is one of the ncclient parameter and seems to be broken.

**Comment by miott7** [comment]
The netconf IP address/port are marked as "optional" in the device profile creation page.  They should be marked as required.

**Comment by einarnn** [comment]
I don't get why a device profile should be required to get a schema?

Also, the "csr" is just shorthand for setting parameters manually. The breakage in the iosxe device handler is fairly obvious; someone obviously copied the CSR device handler but didn't replace the unknown host callback, and left it the same as for the CSR:

        kwargs['unknown_host_cb'] = csr_unknown_host_cb

Cheers,

Einar
